src/llscan.cc

```diff
@@ -289,6 +295,173 @@ bool NodeInfoCmd::DoExecute(SBDebugger d, char** cmd,
 }
 
 
+bool FindReferencesCmd::DoExecute(SBDebugger d, char** cmd,
+                                  SBCommandReturnObject& result) {
+  if (*cmd == NULL) {
+    result.SetError("USAGE: v8 findreferences expr\n");
+    return false;
+  }
+
+  SBTarget target = d.GetSelectedTarget();
+  if (!target.IsValid()) {
+    result.SetError("No valid process, please start something\n");
+    return false;
+  }
+
+  /* Ensure we have a map of objects. */
+  if (!llscan.ScanHeapForObjects(target, result)) {
+    result.SetStatus(eReturnStatusFailed);
+    return false;
+  }
+
+  v8::Value::InspectOptions inspect_options;
+
+  inspect_options.detailed = detailed_;
+
+  char** start = ParseInspectOptions(cmd, &inspect_options);
+
+  // Currently the only parameter is the pointer to search for which
+  // should be a valid lldb expression.
+
+  std::string full_cmd;
+  for (; start != nullptr && *start != nullptr; start++) full_cmd += *start;
+
+  SBExpressionOptions options;
+  SBValue value = target.EvaluateExpression(full_cmd.c_str(), options);
+
+  // Load V8 constants from postmortem data
+  llv8.Load(target);
+
+  if (value.GetError().Fail()) {
+    SBStream desc;
+    if (value.GetError().GetDescription(desc)) {
+      result.SetError(desc.GetData());
+    }
+    result.SetStatus(eReturnStatusFailed);
+    return false;
+  }
+
+  // Check the address we've been given at least looks like a valid object.
+  v8::Value search_value(&llv8, value.GetValueAsSigned());
+  v8::Smi smi(search_value);
+  if (smi.Check()) {
+    result.SetError("Search value is an SMI.");
+    return false;
+  }
+
+  // Walk all the object instances and handle them according to their type.
+  TypeRecordMap mapstoinstances = llscan.GetMapsToInstances();
+  for (auto const entry : mapstoinstances) {
```

Comment:
I think you are probably right - Can I raise a separate issue though? It would probably happen when there are a lot of objects of the same type and the same scenarios would hit findjsinstances in exactly the same way. I can add one consistent option to both the commands at the same time.